### PR TITLE
YM-284 YM-287 | Send GQL query to open-city-profile

### PR DESCRIPTION
**Pull request by charn**
This PR adds logic which is capable of querying the open-city-profile backend.
* `TunnistamoTokenExchange` class is able to exchange an authorization code into an API token for open-city-profile.
* `ProfileAPI` calls the the open-city-profile API and queries myProfile/profile as a user of a given authorization token.

Changes createMyYouthProfile/createYouthProfile so that the profile ID is fetched/verified using Helsinki profile.

### CODE REVIEW

File: common_utils/oidc.py [hunk 1]
```diff
@@ -18,3 +25,97 @@ def authenticate(self, request, **kwargs):
...
+            )
+            session.fetch_token(
+                token_url=oidc_conf["token_endpoint"],
+                authorization_response=redirect_response,
+                client_secret=self.client_secret,
+                include_client_id=True,
+                timeout=self.timeout,
+            )
+        except OAuth2Error:
+            raise TokenExchangeError("Failed to obtain an access token.")
```
Comment (tuomas777) [inline common_utils/oidc.py:86]: Maybe use `raise from` here so that the actual reason isn't lost?

File: common_utils/profile.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        """Fetch profile data for the given profile ID. Requires staff level permission"""
+        query = Template(
+            """
+            query Profile {
+                profile(id: "${id}", serviceType: ${service_type}) {
+                    id
+                }
+            }
+        """
+        ).substitute(id=id, service_type=settings.HELSINKI_PROFILE_SERVICE_TYPE)
```
Comment (tuomas777) [inline common_utils/profile.py:48]: I think the preferred way would be to use GQL variables for the dynamic parts and keep the query static.

**Comment by charn** [comment]
Refactor the mutations to require the profile API token instead of the authorization code. Didn't remove the `TunnistamoTokenExchange` since it could serve a purpose in the GDPR API.

**Review by tuomas777 — APPROVED** [review]
LGTM, great job! 👍 💯 🚀

**Pull request merged**